firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/internal/util/conversions.kt

```diff
@@ -283,8 +283,14 @@ internal fun com.google.ai.client.generativeai.common.GenerateContentResponse.to
   )
 }
 
-internal fun com.google.ai.client.generativeai.common.CountTokensResponse.toPublic() =
-  CountTokensResponse(totalTokens)
+internal fun com.google.ai.client.generativeai.common.CountTokensResponse.toPublic():
+  CountTokensResponse {
+  val billableCharacters = totalBillableCharacters
+  if (billableCharacters != null) {
+    return CountTokensResponse(totalTokens, billableCharacters)
```

Comment:
Any reason you can't do this?

```suggestion
  if (totalBillableCharacters != null) {
    return CountTokensResponse(totalTokens, totalBillableCharacters)
```

Reply:
kotlin compiler complains that it can't do smart casting for `totalBillableCharacters` due to it being part of the public API of a different module:



![Screenshot 2024-04-05 at 7 12 38 AM](https://github.com/firebase/firebase-android-sdk/assets/368578/4a4976ca-f7b8-4854-811c-facfee2e6543)